modules/standard/IO.chpl

```diff
@@ -2401,12 +2396,12 @@ proc openfd(fd: fd_t, hints:iohints=IOHINT_NONE, style:iostyle = defaultIOStyle(
   var err:syserr = ENOERR;
   var ret = openfd(fd, err, hints, style);
   if err {
-    var path:c_string_copy;
+    var path_cs:c_string_copy;
     var e2:syserr = ENOERR;
-    e2 = qio_file_path_for_fd(fd, path);
-    if e2 then path = c"unknown";
-    // FIX ME: could use a toString() that doesn't allocate space
-    ioerror(err, "in openfd", path:string);
+    e2 = qio_file_path_for_fd(fd, path_cs);
+    var path = if e2 then "unknown"
+                     else new string(path_cs, needToCopy=false);
```

Comment:
`c_string_copy`s are not freed when they leave a scope, it is only a renaming of c_string (still a primitive type). I've thought about removing c_string_copy because of this confusion.